runtime/compare.c

```diff
@@ -99,6 +101,22 @@ static intnat compare_val(value v1, value v2, int total)
   return res;
 }
 
+static void poll_pending_actions(struct compare_stack* stk, struct compare_item* sp)
+{
+  if (caml_check_pending_actions()) {
+    value exn;
+    Begin_roots_block((value*)(stk->stack),
+                      (sp - stk->stack) * sizeof(struct compare_stack) / sizeof(value));
```

Comment:
should this be `sizeof(struct compare_item)` rather than `sizeof(struct compare_stack)`?

Reply:
After that "Test 425" in `intext_par` still fails, not really sure why yet.

Reply:
Great, I tried to denote a portion of linear memory and all three of the start address, end address and element size were wrong. Thanks!